### PR TITLE
Test endpoint api open library

### PR DESCRIPTION
**Pull request by maxy96**
# Summary
* Testing of endpoint '/api/open_library/:isbn'
   * With status code 200
       * Response body is not null and with content isbn, title, subtitle, authors, number_of_pages
   * With status code 404
       * Response body with message "book not found" 

# Trello Card
https://trello.com/c/6decFihF/38-3-implementar-fictium-y-crear-test-para-el-endpoint-de-book-information 

# Screenshot
Captura de Pantalla 2021-05-31 a la(s) 12 09 10

### CODE REVIEW

**Comment by maxy96** [comment]
for some reason my * .key files were uploaded

File: app/controllers/api/open_library_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Api
```
Comment (blacksam07) [inline app/controllers/api/open_library_controller.rb:1]: @maxy96 Change the base branch, you already included these changes into another PR # 3

File: spec/request/api_open_library_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'json'
+
+RSpec.describe 'Request to /api/open_library/:isbn', type: :request do
+  describe 'when using ISBN:2' do
```
Comment (blacksam07) [inline spec/request/api_open_library_spec.rb]: the `action`  statement after `describe` define and method that you use in the test

look this 
```ruby
describe PostsController do
  describe action 'GET #index' do # `action` after describe define the method that you make testing
    describe example 'without errors' do # `example` after describe define that is an example of the action
        default_example # You can select an example to be marked as default in your action

        before do
            get :index
        end

        # ... your tests
    end
  end
end

```
```suggestion
  describe action 'Get book info' do
```

File: spec/request/api_open_library_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'json'
+
+RSpec.describe 'Request to /api/open_library/:isbn', type: :request do
```
Comment (blacksam07) [inline spec/request/api_open_library_spec.rb]: this describe is for the class that you make the testing, Controller or Service
```ruby
RSpec.describe OpenLibraryService do
```

```ruby
RSpec.describe Api::OpenLibraryController do
```

File: spec/request/api_open_library_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  describe 'when using ISBN:2' do
+    before { get '/api/open_library/2' }
```
Comment (blacksam07) [inline spec/request/api_open_library_spec.rb]: you need to change, we testing controllers or services or both but not only a request, this way you testing a controller
```ruby
before
  get :show, params: { isbn: isbn }
end

describe example 'with valid value' do
  let(:isbn) { 2 }

  it 'status code 200' do 
    expect(response).to have_http_status(200)
  end

  ...
end
```

**Review by blacksam07 — CHANGES_REQUESTED** [review]

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Fictium.configure do |config|
+  config.fixture_path = File.join(__dir__, 'support', 'docs')
+  config.export_path = File.join(__dir__, '..', 'docs', 'api_docs')
+  config.info.title = 'Books API doc'
+  config.info.description = 'Documentation generated by Wolox - Fictium gem'
+  config.info.version = '1'
+  config.exporters << Fictium::ApiBlueprintExporter.new
+  Fictium.configuration.api_blueprint.host = Rails.application.credentials.config[:base_uri]
+  config.exporters << Fictium::Postman::V2Exporter.new
+  Fictium.configuration.postman.api_url = Rails.application.credentials.config[:base_uri]
```
Comment (blacksam07) [inline spec/rails_helper.rb:77]: you can drop this lines

**Review by blacksam07 — APPROVED** [review]

**Comment by blacksam07** [comment]
@maxy96 fix the conflicts

**Pull request merged**